scripts/linux/buildImage.sh

```diff
@@ -1,56 +1,65 @@
 #!/bin/bash
 
 ###############################################################################
-# This Script builds a specific Edge application in their respective docker
-# containers. This script expects that buildBranch.sh was invoked earlier and
-# all the necessary application files and their Dockerfile be published in
-# directory identified by environment variable BUILD_BINARIESDIRECTORY
+# This script builds an Edge application as a multi-platform Docker image
+# and tags it as:
+#   {registry}/{namespace}/{name}:{version}
+# Each platform-specific image is also tagged:
+#   {registry}/{namespace}/{name}:{version}-{platform}
+# ...where {platform} is one of linux-amd64, linux-arm64v8, or linux-arm32v7.
+#
+# The script expects that buildBranch.sh was invoked earlier and all the
+# application's files were published to the directory '{bin}/publish/{app}',
+# where {bin} is passed to the script's '--bin' parameter and {app} is passed
+# to the '--app' parameter. It also expects that the application's Dockerfile
+# was published to '{bin}/publish/{app}/docker/linux/Dockerfile'.
 ###############################################################################
 
 set -euo pipefail
 
 ###############################################################################
 # Define Environment Variables
 ###############################################################################
-ARCH=$(uname -m)
-SCRIPT_NAME=$(basename "$0")
-PUBLISH_DIR=
-PROJECT=
-DOCKERFILE=
-DOCKER_IMAGENAME=
-DEFAULT_DOCKER_NAMESPACE='microsoft'
-DOCKER_NAMESPACE=${DEFAULT_DOCKER_NAMESPACE}
+APP=
+APP_BINARIESDIRECTORY=
+PLATFORMS='linux/amd64,linux/arm/v7,linux/arm64'
 BUILD_BINARIESDIRECTORY=${BUILD_BINARIESDIRECTORY:=""}
-SKIP_PUSH=0
+DOCKER_IMAGENAME=
+DOCKER_NAMESPACE='microsoft'
+SCRIPT_DIR=$(cd "$(dirname "$0")" && pwd)
+SCRIPT_NAME=$(basename "$0")
 SOURCE_MAP=
 
 ###############################################################################
-# Check format and content of --target-arch argument
+# Check format and content of the --platforms argument
 ###############################################################################
-check_arch() {
-    case "$ARCH" in
-        'x86_64') ARCH='amd64' ;;
-        'armv7l') ARCH='arm/v7' ;;
-        'aarch64') ARCH='arm64' ;;
-        'amd64'|'arm64'|'arm/v7') ;;
-        *) echo "Unsupported architecture '$ARCH'" && exit 1 ;;
-    esac
+check_platforms() {
+    IFS=',' read -a plat_arr <<< "$PLATFORMS"
```

Comment:
this gonna change the current terminal IFS which will interfere with how a subsequential script interprets the `,`.  Please store the current IFS values, then restore such value after the command. 

You can take a look at the sample at: 
https://github.com/Azure/iotedge/blob/main/scripts/linux/smokeTestHelper.sh#L20-L22

Reply:
The IFS variable is being passed into the _command's_ environment, so it doesn't impact the script environment. For example, the following script:

```bash
#!/bin/bash
PLATFORMS='linux/amd64,linux/arm/v7,linux/arm64'
echo "1) IFS: [$IFS]"
IFS=',' read -a plat_arr <<< "$PLATFORMS"
echo "array size: ${#plat_arr[@]}, values: ${plat_arr[@]}"
echo "2) IFS: [$IFS]"
```
returns:

```bash
1) IFS: [ 	
]
array size: 3, values: linux/amd64 linux/arm/v7 linux/arm64
2) IFS: [ 	
]
```
The value of IFS isn't modified.